napalm_junos/junos.py

```diff
@@ -1763,20 +1785,49 @@ def get_optics(self):
         optics_table.get()
         optics_items = optics_table.items()
 
-        # Formatting data into return data structure
-        optics_detail = {}
+        # Format data inserting lane 0 for single lane optics
+        # Format: [(iface, lane, [optical_values]), ...]
+        optics_items_with_lane = []
         for intf_optic_item in optics_items:
+            a = list(intf_optic_item)
+            a.insert(1, u"0")
+            new_intf_optic_item = tuple(a)
+            optics_items_with_lane.append(new_intf_optic_item)
+
+        # Get optical information for 40G/100G optics
+        optics_table40G = junos_views.junos_intf_40Goptics_table(self.device)
+        optics_table40G.get()
+        optics_40Gitems = optics_table40G.items()
+
+        # Format data inserting lane as before.
+        # Format: [(iface, lane, [optical_values]), ...]
+        new_optics_40Gitems = []
+        for item in optics_40Gitems:
+            lane = item[0]
+            iface = item[1].pop(0)
+            new_optics_40Gitems.append((iface[1], unicode(lane), item[1]))
```

Comment:
Remove unicode. For Python 2-3 compatibility, we recommend using https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/utils/py23_compat.py#L12, e.g. `py23_compat.text_type(lane)`. `py23_compat` should already be imported here.